kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemMetadataProvider.java

```diff
@@ -534,6 +535,16 @@ static void writeDescriptor(
         throw new DatasetIOException("Cannot close", e);
       }
     }
+
+    //rename the file into its final location so that changes appear atomic
+      try {
+          fs.rename(tempDescriptorPath, descriptorPath);
```

Comment:
Using the FileContext API, is the rename call still atomic, or does it result in a delete and a rename in the FileSystem API?